Remotion/Core/Core/Context/AsyncLocalStorageProvider.cs

```diff
@@ -0,0 +1,71 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections;
+using System.Threading;
+using Remotion.ServiceLocation;
+using Remotion.Utilities;
+
+namespace Remotion.Context
+{
+  [ImplementationFor (typeof (ISafeContextStorageProvider), Position = 1)]
+  public class AsyncLocalStorageProvider : ISafeContextStorageProvider
+  {
+    private static readonly AsyncLocal<Hashtable> s_data = new (
+        args =>
+        {
+          if (!args.ThreadContextChanged)
+            return;
+
+          if (args.CurrentValue == null)
+            return;
+
+          s_data!.Value = new Hashtable();
```

Comment:
Say, except for using an anonymous function directly in the inline ctor, `s_data` would not be nulalble, correct? would it be cleaner if we used a named method instead?

Reply:
That's true and a nice find.